perception/autoware_tensorrt_bevdet/src/alignbev_plugin.cu

```diff
@@ -0,0 +1,352 @@
+      if (within_bounds_2d(iy_nw, ix_nw, out_H, out_W)) {
+        *out_ptr_NCHW += static_cast<T2>(inp_ptr_NC[iy_nw * out_sH + ix_nw * out_sW]) * nw;
+      }
+      if (within_bounds_2d(iy_ne, ix_ne, out_H, out_W)) {
+        *out_ptr_NCHW += static_cast<T2>(inp_ptr_NC[iy_ne * out_sH + ix_ne * out_sW]) * ne;
+      }
+      if (within_bounds_2d(iy_sw, ix_sw, out_H, out_W)) {
+        *out_ptr_NCHW += static_cast<T2>(inp_ptr_NC[iy_sw * out_sH + ix_sw * out_sW]) * sw;
+      }
+      if (within_bounds_2d(iy_se, ix_se, out_H, out_W)) {
+        *out_ptr_NCHW += static_cast<T2>(inp_ptr_NC[iy_se * out_sH + ix_se * out_sW]) * se;
+      }
+    }
+  }
+}
+
+void create_desc(const int * dims, int nb_dims, TensorDesc & desc)
+{
+  memcpy(&desc.shape[0], dims, sizeof(int) * nb_dims);
+  desc.stride[nb_dims - 1] = 1;
+  for (int i = nb_dims - 2; i >= 0; --i) {
+    desc.stride[i] = desc.stride[i + 1] * desc.shape[i + 1];
+  }
+}
+
+namespace nvinfer1
+{
+// class AlignBEVPlugin
+AlignBEVPlugin::AlignBEVPlugin(const std::string & name) : name_(name)
+{
+}
+
+AlignBEVPlugin::AlignBEVPlugin(const std::string & name, const void * buffer, size_t length)
+: name_(name)
+{
+  memcpy(&m_, buffer, sizeof(m_));
+}
+
+AlignBEVPlugin::~AlignBEVPlugin()
+{
+}
+
+IPluginV2DynamicExt * AlignBEVPlugin::clone() const noexcept
+{
+  auto p = new AlignBEVPlugin(name_, &m_, sizeof(m_));
+  p->setPluginNamespace(namespace_.c_str());
+  return p;
+}
+
+int32_t AlignBEVPlugin::getNbOutputs() const noexcept
+{
+  return 1;
+}
+
+DataType AlignBEVPlugin::getOutputDataType(
+  int32_t index, DataType const * inputTypes, int32_t nbInputs) const noexcept
+{
+  return DataType::kFLOAT;  // FIXME
+}
+
+DimsExprs AlignBEVPlugin::getOutputDimensions(
+  int32_t outputIndex, const DimsExprs * inputs, int32_t nbInputs,
+  IExprBuilder & exprBuilder) noexcept
+{
+  return inputs[0];
+}
+
+bool AlignBEVPlugin::supportsFormatCombination(
+  int32_t pos, const PluginTensorDesc * inOut, int32_t nbInputs, int32_t nbOutputs) noexcept
+{
+  // adj_feat
+  if (pos == 0) {
+    return (inOut[pos].type == DataType::kFLOAT || inOut[pos].type == DataType::kHALF) &&
+           inOut[pos].format == TensorFormat::kLINEAR;
+  } else if (pos == 2) {  // out
+    return (inOut[pos].type == DataType::kFLOAT || inOut[pos].type == DataType::kHALF) &&
+           inOut[pos].format == TensorFormat::kLINEAR;
+  } else if (pos == 1) {  // transform
+    return inOut[pos].type == DataType::kFLOAT && inOut[pos].format == TensorFormat::kLINEAR;
+  }
+  return false;
+}
+
+size_t AlignBEVPlugin::getWorkspaceSize(
+  const PluginTensorDesc * inputs, int32_t nbInputs, const PluginTensorDesc * outputs,
+  int32_t nbOutputs) const noexcept
+{
+  return 0;
+}
+
+int32_t AlignBEVPlugin::enqueue(
+  const PluginTensorDesc * inputDesc, const PluginTensorDesc * outputDesc,
+  const void * const * inputs, void * const * outputs, void * workspace,
+  cudaStream_t stream) noexcept
+{
+  // TODO
```

Comment:
The comment `// TODO` indicates unfinished work within the `enqueue` function.  It is essential to address this TODO before merging the code.  The missing implementation could lead to unexpected behavior or crashes.